app/schemas/tech.relaycorp.gateway.data.database.AppDatabase/1.json

```diff
@@ -2,8 +2,35 @@
   "formatVersion": 1,
   "database": {
     "version": 1,
-    "identityHash": "1eb6cc04dc6143149575f7545f798e49",
+    "identityHash": "1ed41e5e9b3f8b773ce914f529ff07f7",
     "entities": [
+      {
+        "tableName": "Endpoint",
+        "createSql": "CREATE TABLE IF NOT EXISTS `${TABLE_NAME}` (`applicationId` TEXT NOT NULL, `address` TEXT NOT NULL, PRIMARY KEY(`address`, `applicationId`))",
+        "fields": [
+          {
+            "fieldPath": "applicationId",
+            "columnName": "applicationId",
+            "affinity": "TEXT",
+            "notNull": true
+          },
+          {
+            "fieldPath": "address",
+            "columnName": "address",
+            "affinity": "TEXT",
+            "notNull": true
+          }
+        ],
+        "primaryKey": {
+          "columnNames": [
+            "address",
+            "applicationId"
```

Comment:
I think the applicationId shouldn't be part of the PK: Whilst I don't anticipate it'd be a frequent thing, it should be possible to migrate endpoints to another app.

Once we start working on the endpoint implementation, you'll find that each Relaynet service defines the _contract_ to be adhered to by each endpoint participating in the service. Amongst other things, that means Relaynet or its gateways don't care about apps, but endpoints, and -- in turn -- that means that users can migrate their data (inc. endpoint crypto keys) to other compatible apps. 

Say WhatsApp and Signal share the same Relaynet service: People would be able to migrate between those apps without disrupting their communication. However, the actual migration is outside the scope of Relaynet, but it should still support reassigning an endpoint to another app. The team defining the Relaynet-based messaging service should define the export/import process.

Reply:
But how can we notify an app that we got that for them then? I was expecting to use the `applicationId` for that. Without it, the only alternative I had in mind is waking up all relaynet apps with a parameter of the endpoint address we actually wanted to notify.

Reply:
You'd still store the application Id. It just wouldn't be part of the primary key.